src/main/scala/nodes/learning/PCA.scala

```diff
@@ -75,20 +107,50 @@ case class DistributedColumnPCAEstimator(dims: Int) extends Estimator[DenseMatri
  * [[DistributedPCAEstimator]].
  *
  * Automatically decides between distributed and local implementations when node-level optimization is enabled.
+ * The default weights were determined empirically via results run on a 16 r3.4xlarge node cluster.
  *
  * @param dims Dimensions to reduce input dataset to.
+ * @param numMachines
+ * @param cpuWeight
+ * @param memWeight
+ * @param networkWeight
  */
-class OptimizableColumnPCAEstimator(dims: Int) extends OptimizableEstimator[DenseMatrix[Float], DenseMatrix[Float]] {
-  val default = new DistributedColumnPCAEstimator(dims)
+class ColumnPCAEstimator(
+    dims: Int,
+    numMachines: Option[Int] = None,
+    cpuWeight: Double = 3.8e-4,
```

Comment:
We should have a philosophical discussion about whether we want to expose details of optimization at this level of user API, but this is fine for now.